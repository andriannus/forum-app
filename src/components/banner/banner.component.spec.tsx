import { beforeEach, describe, expect, it } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import Banner from "./banner.component";
import type { BannerProps } from "./banner.model";

describe("Banner component", () => {
  beforeEach(cleanup);

  it.each([
    { title: "Masuk" },
    { subtitle: "Silakan masukkan" },
    { title: "Masuk", subtitle: "Silakan masukkan" },
    { center: true, title: "Masuk", subtitle: "Silakan masukkan" },
  ])(
    "should match snapshot with various kinds of props",
    async (props: Partial<BannerProps>) => {
      render(<Banner data-testid="banner" {...props} />);

      const component = screen.queryByTestId("banner");

      expect(component).toMatchSnapshot();
    },
  );

  it.each([{ title: "Masuk" }, { title: "Hallo" }, { title: "Lupa Password" }])(
    "should render `title` correctly based on `title` prop",
    async (props: { title: string }) => {
      render(<Banner data-testid="banner" {...props} />);

      const component = screen.queryByTestId("banner");

      expect(component).toHaveTextContent(props.title);
    },
  );

  it.each([
    { subtitle: "Silakan masukkan" },
    { subtitle: "Selamat Datang di We The Thread" },
  ])(
    "should render `subtitle` correctly based on `subtitle` prop",
    async (props: { subtitle: string }) => {
      render(<Banner data-testid="banner" {...props} />);

      const component = screen.queryByTestId("banner");

      expect(component).toHaveTextContent(props.subtitle);
    },
  );
});
