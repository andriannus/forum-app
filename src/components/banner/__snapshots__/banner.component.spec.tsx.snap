// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Banner component > should match snapshot with various kinds of props 1`] = `
<header
  class="Banner"
  data-testid="banner"
>
  <h1
    class="Banner-title"
  >
    Masuk
  </h1>
</header>
`;

exports[`Banner component > should match snapshot with various kinds of props 2`] = `
<header
  class="Banner"
  data-testid="banner"
>
  <p
    class="Banner-subtitle"
  >
    Silakan masukkan
  </p>
</header>
`;

exports[`Banner component > should match snapshot with various kinds of props 3`] = `
<header
  class="Banner"
  data-testid="banner"
>
  <h1
    class="Banner-title"
  >
    Masuk
  </h1>
  <p
    class="Banner-subtitle"
  >
    Silakan masukkan
  </p>
</header>
`;

exports[`Banner component > should match snapshot with various kinds of props 4`] = `
<header
  class="Banner Banner--center"
  data-testid="banner"
>
  <h1
    class="Banner-title"
  >
    Masuk
  </h1>
  <p
    class="Banner-subtitle"
  >
    Silakan masukkan
  </p>
</header>
`;
